### DISCUSSION TITLE
Question: Working with multiple developers on same database

### DISCUSSION THREAD

**Discussion by RobinC4** [OrchardCMS/OrchardCore#7052] [Q&A] [answered]
Hi,

Just a question, I know the App_data folder contains specific configuration settings for the running website.
But is their a risk when working with multiple developers or "instances" of websites on the same configuration? (We copy the app_data folder to other developers / server).

As currently I had my development environment connected to SQL server database, a published local version to the same database, and a test website also to the same database all with the same settings.
And Not sure what hapenned, but at some point it broke and got an error that IRecipeMigrator could not be injected in OrchardCore.Menu Migrations. While it was running first. Had to remove the orchard tables app-data and create it new without any other changes.

Thanks.

**Comment by mariojsnunes** [comment]
Not sure I understand the entirity of your problem, did you take a look at Migrations?
Imagine you make a change to your database, let's say, turn on a new feature, or create a new ContentType.
You should create a Migration for it. So when other developers get your code, it is run against their database and everything syncs up.

**Comment by deanmarcussen** [comment, marked as answer]
If you are going to work with multiple versions of the website, i.e. multiple developers on the same database you need to be a little bit careful. But without repo steps i can't imagine how this happened
> at some point it broke and got an error that IRecipeMigrator could

Our team has been trialing a dev sql server for the last six months or so.

It's worked well

The procedure for us is 
- Nothing in App_Data (except lucene, but we don't need that for dev much, or can rebuild when we do)
- Everything configured via User Secrets / appsettings.json (docs here https://docs.orchardcore.net/en/dev/docs/reference/core/Configuration/)
- Shells in Database (docs here https://docs.orchardcore.net/en/dev/docs/reference/core/Shells/)
- Content definitions in database
- Data protection in Azure
- Media in Azure
- Sql database in Azure

- I keep the `Sites/Default/appsettings.json` file in git with just the `SqlProvider` string in it to stop setup running when cloning to a new machine.

- Templates are well null checked so when someone changes a content definition and hasn't had time to commit the template it doesn't break anyone else.

- Data to staging servers is done through deployment plans

**Comment by RobinC4** [comment]
Thanks for that.
Yes, migrations we know about and use.
It's more that, for a simple example, I was trying the tenant feature. Which will change the tenants.json file and another site in the app_data.  This is as a development test, but another developer does not get to see the same even though he works on the same database.

It's just a bit about a procedure which information to put where or what to be careful with.

Thanks again.
